docker/scripts/reloader.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * For development purposes only: this script is meant to be run on top
+ * of a Kuzzle starter script. It watches source files (json, ts, js) from
+ * Kuzzle core, plugins, configuration files.
+ * When a file changes, it automatically shuts the current process down, WAIT
+ * until it has correctly been stopped (or kills it after some delay), and only
+ * then restarts it.
+ *
+ * This script is meant to replace solutions like nodemon, which have a few
+ * cumbersome caveats.
+ *
+ * Usage: node reloader.js <config file> <node args> <starter script> <args>
+ *
+ * With:
+ *   - <config file>: this script configuration file
+ *   - <node args>: arguments to pass to the node interpreter
+ *   - <starter script>: the script to execute using node
+ *   - <args>: script arguments
+ */
+
+/* eslint-disable no-console */
+
+'use strict';
+
+const path = require('path');
+const { fork } = require('child_process');
+
+const chokidar = require('chokidar');
+const clc = require('cli-color');
+
+const config = {
+  cwd: '/var/app',
+  killDelay: 5000,
+  watch: [
+    'lib',
+    'plugins',
+    'node_modules',
+    'index.js',
```

Comment:
It should watch `index.ts` as well
```suggestion
    'index.*s',
```

Reply:
why should we restart kuzzle on changes made to index.ts?
it isn't used until recompiled, isn't it?

Reply:
Since with use Node with the `ts-node` package, Typescript files are compiled and interpreted on the fly in the memory so only `index.ts` is really used in development mode

Reply:
ohh, I didn't know that!

Adding this file to the watched list right away :+1: